code/game/objects/items/weapons/tools/_tools.dm

```diff
@@ -633,6 +633,8 @@
 	precision = initial(precision)
 	suitable_cell = initial(suitable_cell)
 	max_fuel = initial(max_fuel)
+	if (reagents)
+		reagents.maximum_volume = max_fuel
```

Comment:
I think it's better to update reagents at the end of this proc to proper handle upgrades and simplify it code.

Reply:
done